File: lib/go/rpc/Connection.go
package rpc

import (
	"bytes"
	"net"
	"sync"
)

type connection struct {
	proto    IProtocol
	trans    ITransport
	conn     *net.TCPConn
	sendCh   chan *bytes.Buffer
	IsValid  bool
	exitSync sync.WaitGroup
	event    *NetEvent
	cb       commonMsgCB
}

func NewConnection(event *NetEvent, ep endPoint, tcpConn *net.TCPConn, proto IProtocol, trans ITransport) *connection {
	conn := &connection{event: event, proto: proto, trans: trans, conn: tcpConn, sendCh: make(chan *bytes.Buffer, 1024), IsValid: true}
	trans.start(conn)
	return conn
}

func (this *connection) GetProtocol() IProtocol {
	return this.proto
}
func (this *connection) GetTransport() ITransport {
	return this.trans
}

File: lib/ts/rpc/Connection.ts
import {Protocol} from "./Protocol"
import {Transport} from "./Transport"
import {ResponseMsg} from "./ResponseMsg"


export abstract class Connection {
	private trans!:Transport;
	private proto!:Protocol;

	protected endPoint:string
	public getProtocol():Protocol    {return this.proto;}
	public getTransport():Transport  {return this.trans;}

	constructor(endPoint:string,trans:Transport,proto:Protocol){
		this.endPoint=endPoint
		this.trans=trans
		this.proto=proto
	}

	public abstract handleInput(buff:Uint8Array):boolean
	public abstract handleClose(ev: Event):boolean
}